lightning/src/routing/router.rs

```diff
@@ -968,42 +977,82 @@ impl cmp::PartialOrd for RouteGraphNode {
 ///
 /// Used to construct a [`PathBuildingHop`] and to estimate [`EffectiveCapacity`].
 #[derive(Clone, Debug)]
-enum CandidateRouteHop<'a> {
+pub enum CandidateRouteHop<'a> {
 	/// A hop from the payer, where the outbound liquidity is known.
 	FirstHop {
+		/// The channel details of the first hop.
+		/// `get_outbound_payment_scid` is assumed to always return Some(scid) - this assumption is checked in
+		/// `find_route` method.
 		details: &'a ChannelDetails,
+		/// The node id of the payer.
+		/// Can be accessed via `source` method.
+		node_id: NodeId
 	},
 	/// A hop found in the [`ReadOnlyNetworkGraph`], where the channel capacity may be unknown.
 	PublicHop {
+		/// The channel info of the hop.
 		info: DirectedChannelInfo<'a>,
+		/// The short_channel_id of the channel.
 		short_channel_id: u64,
+		/// The node id of the current hop in route.
+		source_node_id: NodeId,
+		/// The node id of next hop in route.
+		target_node_id: NodeId,
 	},
 	/// A hop to the payee found in the BOLT 11 payment invoice, though not necessarily a direct
 	/// channel.
 	PrivateHop {
+		/// Hint provides information about a private hop, needed while routing through a private
+		/// channel.
 		hint: &'a RouteHintHop,
+		/// The node id of the next hop in route.
+		target_node_id: NodeId
 	},
 	/// The payee's identity is concealed behind blinded paths provided in a BOLT 12 invoice.
 	Blinded {
+		/// Hint provides information about a blinded hop, needed while routing through a blinded
+		/// channel.
+		/// `BlindedPayInfo` provides information needed about the
+		/// payment while routing through a blinded
+		/// path.
+		/// `BlindedPath` is the blinded path to the destination.
 		hint: &'a (BlindedPayInfo, BlindedPath),
+		/// The index of the hint in the original list of blinded hints.
+		/// Provided to uniquely identify a hop as we are
+		/// route building.
 		hint_idx: usize,
+		/// The node id of the payer.
+		source_node_id: NodeId,
```

Comment:
I think in the `Blinded` and `OneHopBlinded` the source is always `introduction_node_id`, not sure why this is added